tests/resources/plugInfo.json

```diff
@@ -10,9 +10,9 @@
                     }
                 }
             },
-            "LibraryPath": "../libusdOpenAssetIOResolver.so",
+            "LibraryPath": "../../build/dist/libusdOpenAssetIOResolver.so",
```

Comment:
[question] How is this normally managed in other USD plugins? Should we be copying `pluginInfo.json` into `dist` and loading that instead? Then you have a stable path and a self-contained distribution?

Reply:
My understanding is that the `plugInfo.json` isn't something that would go into a final distributable package, other than perhaps as an example, since it's application-specific.

One alternative would be to install (to `dist`) `plugInfo.json`, but only if tests are enabled.

We also have the option to make this a template file (e.g. `plugInfo.json.in`) and render it (e.g. CMake's `configure_file` command) on-demand, replacing the `LibraryPath` with the path to our (installed) plugin.

Reply:
@elliotcmorris I thought these were shipping files, as they identify the entry point to the plugin, otherwise it won't load?

Reply:
It's a bit up in the air, David and I had a discussion about this and tried to work through the workflows of how a distribution would work, and came to the conclusion that most, if not all, situations would end up placing the library file somewhere unknowable to the distribution, even with relative pathing taken into account.

With the thing we noted in the turretResolver example though (ie, the use of CMAKE substitution to distribute this file,) I think there might be some more thinking to do. I think i'd rather make an issue to determine how to distribute the plugin, this file in particular, if that's okay with y'all.

Reply:
Sorry I may be missing something, but is the distribution not a directory, and if you had the json at the root of that dir, and it has the relative path to the lib in, you can put it anywhere you like, and just add that json file to the USD search path? Where does the need for subs come in? Is `LibraryPath` not relative to the json file, rather than `cwd`?

Reply:
My original understanding was that there was only one `plugInfo.json` per application, but looking at [the docs](https://graphics.pixar.com/usd/dev/api/class_plug_registry.html#plug_plugInfo) 

> Plugins are registered by providing a path or paths to JSON file*s*

i.e. plural.

The docs for `plugInfo.json` also show an `Includes` section that can be used to compose multiple JSON files, which is potentially another way that an application can refer to multiple `plugInfo.json` files from a top-level application-specific one.

The docs also talk about relative paths inside `plugInfo.json`, but don't say what they are relative to, which is annoying, but that's a different issue.

So indeed it does look like bundling our `plugInfo.json` is the correct thing to do.

We could still templatise it and use `configure_file` to set the correct (relative) path at install time. Totally optional, but adds flexibility, and flags that a raw `plugInfo.json.in` cannot be used until it is rendered alongside the built plugin .so/.dll.

Reply:
Now installing `pluginfo` to the `dist` directory, 2 fixups.

https://github.com/OpenAssetIO/usdOpenAssetIOResolver/pull/3/commits/09e7e029dd6a58cba36b192c8917cc476c3694cf - Update Cmake to copy file

https://github.com/OpenAssetIO/usdOpenAssetIOResolver/pull/3/commits/4c9f079d0c01a3ac9e8b8ba0bb9c02c1d4ecb1e8 - Update tests to reference file at new location.

Rebase for this is gonna be tricky I can smell it.

Reply:
We can just squash it if need be